File: Easy/problem121.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int min = prices[0];
        int max = prices[0];
        int diff = 0;
        for (int i = 1; i < prices.size(); i++) {
            if (prices[i] < min) {
                min = prices[i];
                max = prices[i];
            }
            if (prices[i] > max) {
                max = prices[i];
            }
            if (max-min>diff) {
                diff = max - min;
            }
        }
    
        return diff;
    }
};

File: Easy/problem121.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        i = len(prices) -1
        maxValue = 0
        maxProfit = 0
        while i >= 0:
            if prices[i] > maxValue:
                maxValue = prices[i]
            if maxValue - prices[i] > maxProfit:
                maxProfit = maxValue - prices[i]
            i-=1
        return maxProfit
